src/frontends/tensorflow/src/op/select.cpp

```diff
@@ -3,35 +3,100 @@
+    // according to the TensorFlow documentation. See in the code:
+    // https://github.com/tensorflow/tensorflow/blob/v2.4.1/tensorflow/lite/kernels/select.cc#L188-L211
+    //
+    // SelectV2 op selects values of 'x' if the corresponding value of 'condition'
+    // is true or the value of 'y' if false. There are valid condition input sizes:
+    // 1. Either the same shape (in which case the select is elementwise), or
+    // 2. Broadcastable shapes between 'condition', 'x' and 'y'.
+    default_op_checks(node, 3, {"SelectV2"});
+    // no preparation for inputs are needed
+    // inputs are already NumPy broadcastable
+    return translate_select_base_op(node, node.get_input(0), node.get_input(1), node.get_input(2));
+}
+
 OutputVector translate_select_op(const NodeContext& node) {
-    TENSORFLOW_OP_VALIDATION(node, node.get_input_size() == 3, "Select op cannot be converted");
-    auto in_1 = node.get_input(0);
-    auto in_2 = node.get_input(1);
-    auto in_3 = node.get_input(2);
-    if (in_1.get_partial_shape().is_static() && in_2.get_partial_shape().is_static()) {
-        // select broadcast
-        if (in_1.get_shape().size() == 1 && in_2.get_shape().size() > 1) {
-            std::vector<uint64_t> axes(in_2.get_shape().size() - 1);
-            std::iota(axes.begin(), axes.end(), 1);
-            auto unsqueeze_axes = make_shared<Constant>(ov::element::i64, Shape{in_2.get_shape().size() - 1}, axes);
-            auto unsqueeze = make_shared<Unsqueeze>(in_1, unsqueeze_axes);
-            auto res = make_shared<Select>(unsqueeze, in_2, in_3);
-            set_node_name(node.get_name(), res);
-            return res->outputs();
-        }
+    // according to the TensorFlow documentation. See in the code:
+    // https://github.com/tensorflow/tensorflow/blob/v2.4.1/tensorflow/lite/kernels/select.cc#L188-L211
+    //
+    // Select op selects values of 'x' if the corresponding value of 'condition' is
+    // true or the value of 'y' if false. There are valid condition input sizes:
+    // 1. Either the same shape (in which case the select is elementwise), or
+    // 2. condition must be Rank 1 and match over the first dimension, or
+    // 3. condition is scalar
+    default_op_checks(node, 3, {"Select"});
+    auto condition = node.get_input(0);
+    auto x = node.get_input(1);
+    auto y = node.get_input(2);
+    auto cond_rank = condition.get_partial_shape().rank();
+    TENSORFLOW_OP_VALIDATION(node,
+                             cond_rank.is_static(),
+                             "TensorFlow Frontend supports Select operation only with condition of static rank.");
+    auto x_rank = x.get_partial_shape().rank();
+    auto y_rank = y.get_partial_shape().rank();
+    TENSORFLOW_OP_VALIDATION(node,
+                             x_rank.is_static() || y_rank.is_static(),
+                             "TensorFlow Frontend supports Select operation only with condition of static rank.");
```

Comment:
I disagree with this approach. Let's take Rank via ShapeOf->ShapeOf and Range to generate unsqueeze axes.

Reply:
I did try this approach. But for the case when axes is empty our `Unsqueeze` does not work:(

Reply:
I propose to create a ticket for Unsqueeze regarding this limitation. Once this limitation is fixed, I will extend this translator.
@jane-intel, how do you think?

Reply:
Agreed to implement this approach using `Reshape`

Reply:
@jane-intel, addressed the comment